pkg/kubectl/stop.go

```diff
@@ -88,12 +90,29 @@ func (reaper *ReplicationControllerReaper) Stop(namespace, name string, timeout
 	if err != nil {
 		return "", err
 	}
+	ctrl, err := rc.Get(name)
+	if err != nil {
+		return "", err
+	}
 	if timeout == 0 {
-		rc, err := rc.Get(name)
-		if err != nil {
-			return "", err
+		timeout = Timeout + time.Duration(10*ctrl.Spec.Replicas)*time.Second
+	}
+
+	// Completely back off till the user has deleted all overlapping controllers. The rc manager
+	// handles overlapping controllers by sorting on creation timestamp and only syncing the
+	// first one. Unless the controller being deleted is also the controller being synced by the
+	// rc manager, stopping it will not converge.
+	overlappingCtrl, err := getReplicationControllersForLabels(rc, labels.Set(ctrl.Spec.Selector))
+	if err != nil {
+		return "", err
+	}
+	if len(overlappingCtrl) > 1 {
+		names := []string{}
+		for _, c := range overlappingCtrl {
+			names = append(names, c.Name)
 		}
-		timeout = Timeout + time.Duration(10*rc.Spec.Replicas)*time.Second
+		return "", fmt.Errorf(
+			"Overlapping controllers detected %v. Please delete all the others with --cascade=false.", strings.Join(names, ","))
```

Comment:
I see, so you're suggesting that we sort the returned list of controllers just like the rc manager would and see if the controller being deleted is at the head of the list, and if it isn't, just skip the scale. correct? This should work and when we introduce a new status.rcState=dormant down the line, we can use that instead of the sort.